src/devices/vdml_usd.cpp

```diff
@@ -29,6 +29,15 @@ std::vector<std::string> list_files(const char* path) {
 	std::vector<std::string> files = {};
 	// malloc buffer to store file names
 	char *buffer = (char *) malloc(10000);
+	if (buffer == NULL) {
+		// try again smaller buffer to see if that works
+		buffer = (char *) malloc(500);
```

Comment:
Keeping a `buffer_size` variable is even more pertinent with this addition, wouldn't want to put 10000 entries into a buffer of size 500! It can't be `constexpr` anymore now though, so ignore that part of the above comment.